src/opl/generator.cpp

```diff
@@ -952,6 +990,60 @@ void Generator::updateRegBD()
     WriteReg(0x0BD, m_regBD);
 }
 
+uint32_t Generator::getChipVolume(
+    uint32_t vol, uint8_t ccvolume, uint8_t ccexpr, int volmodel)
+{
+    uint32_t volume;
+
+    switch(volmodel)
+    {
+    case VOLUME_Linear:
+    {
+        volume = vol * ccvolume * ccexpr;
+        volume = volume * 127 / (127 * 127 * 127) / 2;
+    }
+    break;
+
+    default:
+    case VOLUME_Generic:
+    case VOLUME_CMF:
```

Comment:
The `CMF` is same as `VOLUME_Linear`, but `CMF` is merged with Generic in libADLMIDI due dumb "logarithmic volume" I must remove and the functions map to volume model and return `volumeModel == CMF`

Reply:
Ok